### DISCUSSION TITLE
@file streams-url_raw-serial.ino stop serial when file end

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#302] [Q&A] [answered]
@file streams-url_raw-serial.ino

This example continuously send data to serial plotter . How can we stop when the url file end or complete  ? also can we stop in mid and then resume later ?

**Comment by aqildad-create** [comment]
also can we stop in the mid and then resume later ??

**Comment by pschatzmann** [comment, marked as answer]
Sure. Just make sure that the copy in the loop is not executed if you want pause.
If you are new in programming you can consider to use the AudioPlayer class instead where you can call stop() and play()

**Comment by aqildad-create** [comment]
Is serial data  same as data present in  raw file or its factor to some value in order to plot in serial-PLOTER . 

i want to compare this Serial data with original raw data present in file . . How to get back original data  ?

**Comment by pschatzmann** [comment]
It is just convertering the binary values to a string so that you can read it. If you want to see the binary data instead you can use a HexDumpStream as output instead.
